networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/gossip/AggregateTopicHandler.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.networking.p2p.jvmlibp2p.gossip;
+
+import static tech.pegasys.artemis.datastructures.util.AttestationUtil.get_indexed_attestation;
+import static tech.pegasys.artemis.datastructures.util.AttestationUtil.is_valid_indexed_attestation;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import io.libp2p.core.pubsub.PubsubPublisherApi;
+import io.libp2p.core.pubsub.Topic;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.ssz.SSZException;
+import tech.pegasys.artemis.datastructures.operations.AggregateAndProof;
+import tech.pegasys.artemis.datastructures.operations.IndexedAttestation;
+import tech.pegasys.artemis.datastructures.state.BeaconState;
+import tech.pegasys.artemis.datastructures.util.SimpleOffsetSerializer;
+import tech.pegasys.artemis.storage.ChainStorageClient;
+
+public class AggregateTopicHandler extends GossipTopicHandler<AggregateAndProof> {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private static final Topic TOPIC = new Topic("/eth2/beacon_aggregate_and_proof/ssz");
+  private final ChainStorageClient chainStorageClient;
+
+  protected AggregateTopicHandler(
+      final PubsubPublisherApi publisher,
+      final EventBus eventBus,
+      final ChainStorageClient chainStorageClient) {
+    super(publisher, eventBus);
+    this.chainStorageClient = chainStorageClient;
+  }
+
+  @Override
+  public Topic getTopic() {
+    return TOPIC;
+  }
+
+  @Subscribe
+  public void onNewAggregate(final AggregateAndProof aggregateAndProof) {
+    gossip(aggregateAndProof);
+  }
+
+  @Override
+  protected AggregateAndProof deserialize(final Bytes bytes) throws SSZException {
+    return SimpleOffsetSerializer.deserialize(bytes, AggregateAndProof.class);
+  }
+
+  @Override
+  protected boolean validateData(final AggregateAndProof aggregateAndProof) {
+    // TODO: implement the is_aggregator validation logic
+    //  (needs to have is_aggregator and slot_signature logic implemented)
+    final BeaconState state =
+        chainStorageClient
+            .getStore()
+            .getBlockState(aggregateAndProof.getAggregate().getData().getBeacon_block_root());
```

Comment:
I had taken this code directly from the AttestatinTopicHandler so the same problem exists there. Since it's not in the scope of this PR, I made a Jira issue for it: https://pegasys1.atlassian.net/browse/BC-118

Reply:
good catch - I noticed this before but forgot to fix it 🤦‍♀